tests/util.py

```diff
@@ -14,8 +14,11 @@ def clean_xml(xml_str):
     Remove whitespace from XML.
     """
     parser = etree.XMLParser(remove_blank_text=True)
-    return etree.tostring(
-        etree.XML(xml_str, parser=parser))
+    try:
+        return etree.tostring(
+            etree.XML(xml_str, parser=parser)).decode('utf-8')
+    except UnicodeError:
```

Comment:
It is not thrown in the tests.

I'm assuming that since you wanted this code to not just typecast to string
and instead specifically encode it as utf-8, you wanted it to be resilient
to these sorts of failures. You also specifically mention catching raised
exceptions in the last comment you made about this (AttributeError in
particular).

Honestly, this is just test utility code, so I don't have a strong
preference on it being super performance focused or resilient. I'm just
trying to make the changes you want.

On Tue, Oct 20, 2015 at 10:53 AM, George Schneeloch <
notifications@github.com> wrote:

> In tests/util.py
> <https://github.com/mitodl/xbundle/pull/48#discussion_r42529753>:
>
> > @@ -14,8 +14,11 @@ def clean_xml(xml_str):
> >      Remove whitespace from XML.
> >      """
> >      parser = etree.XMLParser(remove_blank_text=True)
> > -    return etree.tostring(
> > -        etree.XML(xml_str, parser=parser))
> > +    try:
> > +        return etree.tostring(
> > +            etree.XML(xml_str, parser=parser)).decode('utf-8')
> > +    except UnicodeError:
>
> Why do we need to catch this exception? Are we encountering it in the
> tests?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mitodl/xbundle/pull/48/files#r42529753>.
>